reference/sql_workarounds.rst

```diff
@@ -0,0 +1,145 @@
+.. _workarounds:
+
+SQL Workarounds
+===============
+
+Citus supports most, but not all, SQL statements directly. Its SQL support continues to improve. Also many of the unsupported features have workarounds; below are a number of the most useful.
+
+Subqueries
+----------
+
+In analytics it is common to model information using a "star schema." This is a collection of dimensional tables (e.g. observations, statistics) referencing a table of facts (e.g. people, pages or events). Furthermore, a common form of query examines dimensions for a subset of the facts. Examples of these kind of queries:
+
+* "Get aggregated statistics about football players whose passes exceed three hundred yards."
+* "Which other pages do people visit who view a certain page?"
+* "Count the number of distinct sessions for the top twenty-five most visited blog posts."
+* "What is the number of people in companies among the top fifty product areas?"
+
+The general form of the query in SQL is
+
+.. code-block:: sql
+
+  select dim_a, dim_b, dim_c
+    from dimensions
+   where fact_id in (select id from facts where ...conditions...)
+     and ...conditions...;
+
+Citus does not allow subqueries in the WHERE clause so we must choose a workaround.
+
+Workaround 1. Generate explicit WHERE-IN expression
```

Comment:
I thought this version is perhaps superior to the inlining technique because the latter takes more time to stream results to the client whereas this one all happens in the db.